pkg/generate/app/sourcelookup.go

```diff
@@ -378,7 +381,7 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		Binary: repo.binary,
 	}
 
-	if repo.Info() != nil && repo.Info().Dockerfile != nil {
+	if (repo.ignoreRepository || repo.forceAddDockerfile) && repo.Info() != nil && repo.Info().Dockerfile != nil {
```

Comment:
> "dockerfile" tells me nothing.

Hmm it's not "dockerfile" for some hours already :)

Take a fresh look, it's been called "forceAddDockerfile".